addons/sale/controllers/portal.py

```diff
@@ -307,9 +308,11 @@ def portal_order_transaction(self, order_id, access_token, **kwargs):
         except AccessError:
             raise ValidationError(_("The access token is invalid."))
 
+        logged_in = not request.env.user._is_public()
+        partner = request.env.user.partner_id if logged_in else order_sudo.partner_invoice_id
         kwargs.update({
             'reference_prefix': None,  # Allow the reference to be computed based on the order
-            'partner_id': order_sudo.partner_invoice_id.id,
+            'partner_id': partner.id,
```

Comment:
Nothing to say to this line here, but there are so many changes done to kwargs that I would change the docstring. "Locally unused data passed to _create_transaction" is really misleading IMO.

Reply:
Hm, I'm not sure, because they are still unused as in we don't really do anything with the values that we receive we just override/add to kwargs.